spec/models/polling/cert_status_updater_spec.rb

```diff
@@ -0,0 +1,130 @@
+        expect(sp_enc_cert_for_multiple_entities.in_use_at).to be(nil)
+      end
+
+      it "updates the cert's timestamp if a call to the Hub shows it is in use for all relevant entity IDs" do
+        hub_config_api = double('hub_config_api')
+        allow(hub_config_api).to receive(:encryption_certificate).with('entity 1').and_return(sp_enc_cert_for_multiple_entities.value)
+        allow(hub_config_api).to receive(:encryption_certificate).with('entity 2').and_return(sp_enc_cert_for_multiple_entities.value)
+        allow(hub_config_api).to receive(:encryption_certificate).with('entity 3').and_return(sp_enc_cert_for_multiple_entities.value)
+
+        cert_status_updater = CertStatusUpdater.new(hub_config_api)
+
+        expect(sp_enc_cert_for_multiple_entities.in_use_at).to be(nil)
+        cert_status_updater.update_hub_usage_status_for_cert(sp_enc_cert_for_multiple_entities)
+        expect(sp_enc_cert_for_multiple_entities.in_use_at).not_to be(nil)
+      end
+    end
+
+    context 'When given a signing cert belonging to an SP component' do
+      it "leaves the cert unchanged if a call to the Hub does not show it is in use for all relevant entity IDs" do
+        hub_config_api = double('hub_config_api')
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 1').and_return(sp_sgn_cert_for_multiple_entities.value, sp_sgn_cert_other.value)
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 2').and_return(sp_sgn_cert_for_single_entity.value, sp_sgn_cert_other.value)
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 3').and_return(sp_sgn_cert_for_multiple_entities.value)
+
+        cert_status_updater = CertStatusUpdater.new(hub_config_api)
+
+        expect(sp_sgn_cert_for_multiple_entities.in_use_at).to be(nil)
+        cert_status_updater.update_hub_usage_status_for_cert(sp_sgn_cert_for_multiple_entities)
+        expect(sp_sgn_cert_for_multiple_entities.in_use_at).to be(nil)
+      end
+
+      it "updates the cert's timestamp if a call to the Hub shows it is in use for all relevant entity IDs" do
+        hub_config_api = double('hub_config_api')
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 1').and_return(sp_sgn_cert_for_multiple_entities.value, sp_sgn_cert_other.value)
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 2').and_return(sp_sgn_cert_for_multiple_entities.value, sp_sgn_cert_other.value)
+        allow(hub_config_api).to receive(:signing_certificates).with('entity 3').and_return(sp_sgn_cert_for_multiple_entities.value)
+
+        cert_status_updater = CertStatusUpdater.new(hub_config_api)
+
+        expect(sp_sgn_cert_for_multiple_entities.in_use_at).to be(nil)
+        cert_status_updater.update_hub_usage_status_for_cert(sp_sgn_cert_for_multiple_entities)
+        expect(sp_sgn_cert_for_multiple_entities.in_use_at).not_to be(nil)
+      end
+    end
+  end
```

Comment:
can we have one more test where the certificate's component doesn't have a service?
just to exercise test for this line `certificate.component.services.map(&:entity_id)` (in `cert_status_updater.rb`) .. to make sure it doesn't blow up when a service hasn't been added to the component (yet)

Reply:
Good idea; I'll dd this.